src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -9035,6 +9039,10 @@ bool StorageReplicatedMergeTree::removeSharedDetachedPart(DiskPtr disk, const St
 
     NameSet files_not_to_remove;
 
+    auto part_info = DetachedPartInfo::parseDetachedPartName(disk, part_name, MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING);
```

Comment:
Maybe we need to add a comment why is it safe to always use `MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING`.

Reply:
I'll add a comment. The same constant is already used in this [method a few lines](https://github.com/ClickHouse/ClickHouse/pull/48862/files#diff-0f4308161ab179a68f01cabfce4dc9e6e4e5f9c1e6c46c37d5adaeaad74e9913R9060) below, so we are not introducing any new assumptions here.